File: src/interfaces/IMovePacer.ts
// Implemented by whatever presents the cube (e.g a web view or a robot). After the Solver
// makes a move on the engine it awaits settled(), giving the presentation time to present
// that move before the next one. This is the only thing the solver knows about the outside
// world — it never references a specific style of presentation.
export interface IMovePacer {
  // Resolves once the presentation has finished presenting the latest move (animation
  // or motor movement complete). A presentation may reject the Promise to signal a failure.
  settled(): Promise<void>
}

File: android/core/src/main/kotlin/com/mikeb/simplepuzzlecube/interfaces/IMovePacer.kt
// Port of ../src/interfaces/IMovePacer.ts.
package com.mikeb.simplepuzzlecube.interfaces

// The one thing the solver knows about the outside world: a pacer whose `settled()`
// resumes once the presentation has fully presented the latest move (instantly for a
// headless run, animation-end for a view, motor completion for a robot). TS
// `settled(): Promise<void>` → a suspend fun; a rejected promise → a throw.
interface IMovePacer {
    suspend fun settled()
}
